dist-build/lang-model-tck-runner/src/test/java/org/jboss/weld/lang/model/tck/BuildtimeVetoExtension.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.weld.lang.model.tck;
+
+import jakarta.annotation.Priority;
+import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.inject.spi.Extension;
+import jakarta.enterprise.inject.spi.ProcessAnnotatedType;
+
+/**
+ * Vetos all classes for the build-time Lang Model TCK extension to avoid deployment issues with unaccessible package private members.
+ * @author Ondro Mihalyi
+ */
+public class BuildtimeVetoExtension implements Extension {
```

Comment:
A cleaner solution might be to change `LangModelTckTest` to use `annotated` discovery instead of `all`?
Another alternative is to change the TCK package and add `package-info.java` with `@Vetoed` in it.
Here is an example of what I mean from Weld code - https://github.com/weld/core/blob/master/impl/src/main/java/org/jboss/weld/bean/package-info.java

Reply:
None of these solutions would work because they would disable the `BuildCompatibleExtension` or one of its dependencies completely and the TC would fail. The solution in this PR is more of a hack but just works - the `BuildCompatibleExtension` would be loaded before the `ProcessAnnotatedType` in order to pass the TCK but then all its classes are vetoed so that they aren't processed later. If they are processed by WildFly later, an exception is thrown because some of the classes are package private and not available via reflection.
I think the only clean solutions would be either to veto each individual class in the CDI that is package private or make those classes public.

Reply:
It would work, the test relies on `LangModelVerifier` being discovered as a type (while not having any bean defining annotation). You can still set discovery to `annotated` and add the following method to the `BuildCompatibleExtension`:

```
    @Discovery
    public void addVerifier(ScannedClasses sc) {
        sc.add(LangModelVerifier.class.getName());
    }
```

With that, you should be able to remove `BuildtimeVetoExtension`. The reason I am trying to do that is because with it, the example isn't valid for any CDI Lite implementation trying to execute this TCK test. Lite impls do not have access to portable extension.

Reply:
You're right, that worked. I've updated this PR along what you suggested.